source/expressions.tex

```diff
@@ -456,15 +456,15 @@
 Using an overloaded operator causes a function call; the
 above covers only operators with built-in meaning.
 \end{note}
-If the (possibly converted) expression is a prvalue,
-the temporary materialization conversion\iref{conv.rval} is applied.
+The temporary materialization conversion\iref{conv.rval} is applied
+if the (possibly converted) expression is a prvalue of object type.
+The expression is evaluated and its result (if any) is discarded.
 \begin{note}
 If the expression is an lvalue of
 class type, it must have a volatile copy constructor to initialize the
 temporary object that is the result object of the lvalue-to-rvalue
 conversion.
 \end{note}
```

Comment:
This note is quite distant from "the expression" that it's talking about, which is the original discarded-value expression prior to any conversions. Can we get it a bit closer to the wording above that talks about introducing an lvalue-to-rvalue conversion?

Reply:
But the temporary materialization conversion is the one that actually creates the object (and needs the volatile copy constructor).

Suggestion: Say "If the original expression" and replace "lvalue-to-rvalue conversion" with "temp. mat. conv." and move "The expression is evaluated and its result (if any) is discarded." to after the note.